bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/gsaoi/Gsaoi.java

```diff
@@ -726,24 +745,59 @@ private static boolean isExpensiveOffset(Config cur, Option<Config> prev) {
         return curGuided || isGuided(prev);
 
     }
+    // This is for use in the ITC overheads calculations only.
+    // LGS reacquisition is required when coming back from sky offset >5'
+    private static boolean lgsReacquisitionRequired(Config cur, Option<Config> prev) {
+        if (!isOffset(cur, prev)) {
+            return false;
+        }
 
-    // REL-1103
+        if (prev.isDefined()) {
+            Offset curOff  = OffsetOverheadCalculator.instance.extract(cur);
+            Offset prevOff = OffsetOverheadCalculator.instance.extract(prev);
+            double distance = curOff.distance(prevOff).toArcsecs().getMagnitude();
+
+            if (isGuided(cur) && !isGuided(prev)) {
+
+                if (distance > 300.0) {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+        // REL-1103
     // Get correct offset overhead in the common group.  If a guided offset
     // or a switch from guided to non-guided, it is expensive.  If going from
     // a sky position to another sky position, it counts as a normal offset.
     private CategorizedTimeGroup commonGroup(Config cur, Option<Config> prev) {
         CategorizedTimeGroup ctg = CommonStepCalculator.instance.calc(cur, prev);
-        return (isExpensiveOffset(cur, prev)) ?
-            ctg.filter(RM_OFFSET_OVERHEAD).add(GUIDED_OFFSET_OVERHEAD_CATEGORIZED_TIME) :
-            ctg;
+
+        // This is used only for the ITC overhead calculations, since in the OT the sky
+        // observations with large offsets are made into separate observations
+        if (lgsReacquisitionRequired(cur,prev)) {
+            ctg = ctg.add(LGS_REACQUISITION_OVERHEAD_CATEGORIZED_TIME);
+        }
+        if (isExpensiveOffset(cur, prev)) {
+            ctg = ctg.filter(RM_OFFSET_OVERHEAD).add(GUIDED_OFFSET_OVERHEAD_CATEGORIZED_TIME);
+        }
+        return ctg;
     }
 
 
+    public double readout(int coadds, int lowNoiseReads) {
+        return 21 + 2.8 * lowNoiseReads * coadds + 6.5 * (coadds - 1);
+    }  // REL-445
+
     @Override public CategorizedTimeGroup calc(Config cur, Option<Config> prev) {
```

Comment:
This method is making me really uncomfortable, because it's declared in two interfaces (`StepCalculator` and `ItcOverheadsProvider`) that are both inherited by this class, and some others.

Having the same method coming from two interfaces seems like a design flaw.